src/panels/config/backup/dialogs/dialog-restore-backup.ts

```diff
@@ -280,26 +281,36 @@ class DialogRestoreBackup extends LitElement implements HassDialog {
   }
 
   private _subscribeBackupEvents() {
-    this._unsub = subscribeBackupEvents(this.hass!, (event) => {
-      if (event.manager_state === "idle" && this._state === "in_progress") {
-        this.closeDialog();
-      }
-      if (event.manager_state !== "restore_backup") {
-        return;
-      }
-      this._state = event.state;
-      if (event.state === "completed") {
-        this.closeDialog();
-      }
-      if (event.state === "failed") {
-        this._error = this.hass.localize(
-          "ui.panel.config.backup.dialogs.restore.restore_failed"
-        );
-      }
-      if (event.state === "in_progress") {
-        this._stage = event.stage;
+    this._unsub = subscribeBackupEvents(
+      this.hass!,
+      (event) => {
+        if (event.manager_state === "idle" && this._state === "in_progress") {
+          this.closeDialog();
+        }
+        if (event.manager_state !== "restore_backup") {
+          return;
+        }
+        this._state = event.state;
+        if (event.state === "completed") {
+          this.closeDialog();
+        }
+        if (event.state === "failed") {
+          this._error = this.hass.localize(
+            "ui.panel.config.backup.dialogs.restore.restore_failed"
+          );
+        }
+        if (event.state === "in_progress") {
+          this._stage = event.stage;
+        }
+      },
+      async () => {
+        if (isComponentLoaded(this.hass, "backup")) {
+          return true;
+        }
+        return (await waitForIntegrationSetup(this.hass, "backup"))
+          .integration_loaded;
```

Comment:
This seems redundant but I think it can be false if the integration setup fails

Reply:
yes 👍